gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java

```diff
@@ -469,7 +477,9 @@
         .setEncryptionAlgorithm(GCS_ENCRYPTION_ALGORITHM.get(config, config::get))
         .setEncryptionKey(RedactedString.create(GCS_ENCRYPTION_KEY.getPassword(config)))
         .setEncryptionKeyHash(RedactedString.create(GCS_ENCRYPTION_KEY_HASH.getPassword(config)))
-        .setGrpcEnabled(GCS_GRPC_ENABLE.get(config, config::getBoolean));
+        .setGrpcEnabled(GCS_GRPC_ENABLE.get(config, config::getBoolean))
+        .setAuthorizationProvider(GCS_AUTHORIZATION_PROVIDER.get(config, config::get))
```

Comment:
Instead of a string, I think we should put the instance of the class that implements the authorization handler in the GoogleCloudStorageOptions class itself. You would do reflection here.

Something like:
```
String impl = GCS_AUTHORIZATION_HANDLER_IMPL.get(config, config::get));
if (impl.isNotEmpty()) {
    AuthorizationHandler authorizationHandler = ... reflection logic here ...
    authorizationHandler.setConf(config);
    builder.setAuthorizationHandler(authorizationHandler)
}
```